Cron/Images.php

```diff
@@ -0,0 +1,236 @@
+            ->setPageSize(100);
+        // iterate over the pages:
+        $eans = array();
+        $currentPage = 0;
+        $lastPage = $collection->getLastPageNumber();
+        while ($currentPage < $lastPage) {
+            // load the data of this page in a single query:
+            $collection->setCurPage(++$currentPage);
+            $collection->load();
+            // iterate over the products:
+            foreach ($collection as $product) {
+                if ($ean = $product->{$attribute}) {
+                    // add the ean if this product has one:
+                    $eans[] = $ean;
+                }
+            }
+        }
+        return $eans;
+    }
+
+    /**
+     * Saves images array from stockbase for given $ean
+     *
+     * @param array $images
+     *
+     * @return bool
+     */
+    public function saveImageForProduct($images)
+    {
+        // get product model:
+        $productModel = $this->product;
+        // get ean attribute:
+        $eanField = $this->config->getEanFieldName();
+        // get client:
+        $client = $this->stockbaseClientFactory->create();
+        // loop images:
+        foreach ($images as $image) {
+            // load product by ean:
+            $product = $productModel->loadByAttribute($eanField, $image->EAN);
+            // continue looping if we do not have product:
+            if (!$product) {
+                continue;
+            }
+            // get image from stockbase:
+            $protectedImage = $client->getImageFile($image->{'Url'});
+            // create temporal folder if it is not exists:
+            $tmpDir = $this->getMediaDirTmpDir();
+            $this->file->checkAndCreateFolder($tmpDir);
+            // get new file path:
+            $newFileName = $tmpDir . baseName($image->{'Url'});
+            // read file from URL and copy it to the new destination:
+            $result = $this->file->read($protectedImage, $newFileName);
+            if ($result) {
+                if ($product->getMediaGallery() == null) {
+                    $product->setMediaGallery(array('images' => array(), 'values' => array()));
+                }
+                // add saved file to the $product gallery:
+                $product->addImageToMediaGallery(
+                    $newFileName,
+                    array('image', 'small_image', 'thumbnail'),
+                    false,
+                    false
+                );
+                // save product:
+                $product->{'save'}();
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Media directory name for the temporary file storage
+     * pub/media/tmp
+     *
+     * @return string
+     */
+    protected function getMediaDirTmpDir()
+    {
+        return $this->directoryList->getPath(DirectoryList::MEDIA) . DIRECTORY_SEPARATOR . 'tmp';
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getImagesEans()
```

Comment:
Please Make private – same for all other methods that are nevere accessed from outside this class

Reply:
Done.